src/Swarm/Game/Value.hs

```diff
@@ -80,8 +82,19 @@ data Value where
   VDelay :: Term -> Env -> Value
   -- | A reference to a memory cell in the store.
   VRef :: Int -> Value
+  -- | A computation running in another thread.
+  VAsync :: GAsync Value -> Value
   deriving (Eq, Show)
 
+data GAsync a = GAsync
+  { asyncIndex :: Int
```

Comment:
What is the `asyncIndex` for?  Just to have a way to `Show` it?

Reply:
I got rid of it, `Async` has hash and `Ord`, that is all I need :slightly_smiling_face: